src/schema-tools/superjson/normalize.ts

```diff
@@ -332,9 +345,27 @@ export function normalizeProviderSettings(
     );
   }
 
+  let normalizedSettings: NormalizedProviderSettings & { ast?: unknown };
+
+  if (isSettingsWithAst(providerEntry)) {
+    // return { ast: providerEntry.ast } as unknown as NormalizedProviderSettings;
+
+    normalizedSettings = {
+      ast: providerEntry.ast,
+      security: providerEntry.security ?? [],
+      parameters: providerEntry.parameters ?? {},
+    };
+  } else {
+    normalizedSettings = {
+      file: providerEntry.file,
+      security: providerEntry.security ?? [],
+      parameters: providerEntry.parameters ?? {},
+    };
+  }
+
   if (environment !== undefined) {
     return {
-      file: providerEntry.file,
+      ...normalizedSettings,
```

Comment:
question: Wouldn't it be better to assign to `security: normalizedSettings.security.map(...resolveEnvRecord...)`? That way even if someone changes `security` and `parameters` in the ifs above it will propagate into the result object.

Reply:
🤦‍♂️definitely yes. Thanks